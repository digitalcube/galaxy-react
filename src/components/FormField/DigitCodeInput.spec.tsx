import { render } from '@testing-library/react';
import React from 'react';
import {
  Default as DigitCodeInput,
  Uncontrolled,
} from './DigitCodeInput.stories';

describe('DigitCodeInput', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <DigitCodeInput {...(DigitCodeInput.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot (Uncontrolled)', () => {
    const { asFragment } = render(
      <Uncontrolled {...(Uncontrolled.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
