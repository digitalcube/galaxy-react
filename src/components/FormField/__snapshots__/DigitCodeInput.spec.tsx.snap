// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DigitCodeInput to match snapshot (Uncontrolled) 1`] = `
<DocumentFragment>
  <div
    class="form-row mx-0 verification-code"
  >
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="d-block invalid-feedback"
    >
      code is incorrect
    </div>
  </div>
</DocumentFragment>
`;

exports[`DigitCodeInput to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-row mx-0 verification-code"
  >
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="col-2 px-0"
    >
      <input
        class="form-control px-0 text-center"
        inputmode="numeric"
        maxlength="1"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
